scripts/schemas/validate.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"cuelang.org/go/cue/load"
+	"github.com/perses/perses-plugins/scripts/npm"
+	"github.com/sirupsen/logrus"
+)
+
+const schemasPath = "backend"
+
+// check that the CUE schemas for a given plugin are valid (= not raising errors)
+func validateSchema(pluginName string) error {
+	// load the cue files into build.Instances slice
+	// package `model` is imposed so that we don't mix model-related files with migration-related files
+	buildInstances := load.Instances([]string{}, &load.Config{Dir: filepath.Join(pluginName, schemasPath), Package: "model"})
+	// we strongly assume that only 1 buildInstance should be returned, otherwise we skip it
+	// TODO can probably be improved
+	if len(buildInstances) != 1 {
+		return fmt.Errorf("the number of build instances is != 1")
+	}
+	buildInstance := buildInstances[0]
+
+	// check for errors on the instances
+	if buildInstance.Err != nil {
+		return buildInstance.Err
+	}
+
+	return nil
+}
+
+func main() {
+	workspaces, err := npm.GetWorkspaces()
```

Comment:
should we create a script/function that can be reused per plugin. Would this be helpful to be included in the plugin template so eternal implementations can also benefit from this?

Reply:
I would keep it this way. A plugin without the frontend part cannot work. So getting the list of the npm packages to list the plugin here is ok from my opinion.

By looking at the (already big) list of the plugins, I am wondering if we shouldn't create a sub folder `panel`. Just to avoid having everything at the root and not able to know by a quick look where are the panels / datasources / queries ...etc.

Reply:
> should we create a script/function that can be reused per plugin. Would this be helpful to be included in the plugin template so eternal implementations can also benefit from this?

Yeah that would make sense definitely. I would add it in the CLI for sure